pkg/cmd/generate/cluster.go

```diff
@@ -67,3 +70,13 @@ func ensureUsers(ctx *clusterContext, objsCache objectsCache) error {
 
 	return nil
 }
+
+var specialCharRegexp = regexp.MustCompile("[^A-Za-z0-9]")
+
+func sanitizeUserName(userName string) string {
```

Comment:
before you start asking why I didn't use the already existing functions, I was thinking about it, but decided to use this simple option, the reasons are:
* **transforming username from toolchain-common** - the usage is different, it cuts the name of the username (to keep it short) as it expects that it would be used together with the namespace/space suffixes. It also does some assumptions with `@` char.
* **transforming usersignup name that is in registration service** - yes, we could use this one, but we would need to move it to toolchain-common first. However, I thought that we don't need any extra logic of generating the sha in case there is some special char in the username. TBH, the risk of having a conflict of the names is very low, so I would rather keep it simple and let the transformation that is in reg-service fully optimized for the usage for UserSignup names.

Reply:
Thanks for this explanation :D